### DISCUSSION TITLE
Blazor App CSLA 6.0 Business object with multiple parameters

### DISCUSSION THREAD

**Discussion by jhuerta3** [MarimerLLC/csla#3085] [Questions] [answered]
Hello,

I was hoping someone can help me out. I just started working on a new Blazor app and I have been trying to figure out how to get a business object with more than one parameter. In the past we used CriteriaBase but I’ve read that its no longer being used and I cant seem to figure out the method signatures to do it. This is what I’m trying to do. I’m also not entirely sure what the IEmployeeDal interface should look like. Any assistance would be appricated. I done it with pass version of CSLA but I cant find my samples. 

        [Fetch]
        private void Fetch(Guid EmployeeID, [Inject] DataAccess.IEmployeeDal dal)
        {
            var data = dal.Get(EmployeeID);
            using (BypassPropertyChecks)
                Csla.Data.DataMapper.Map(data, this);
            BusinessRules.CheckRules();
        }

        [Fetch]
        public static EmployeeEdit GetEmployeeByName(string firstname, string lastname)
        {
            var criteria = new Criteria();
            criteria.FirstName = firstname;
            criteria.LastName = lastname;
            return DataPortal.Fetch<EmployeeEdit>(criteria);
        }

**Comment by JacoJordaan** [comment]
It should look be something like:
````
    [Fetch]
    private void Fetch(Guid EmployeeID, [Inject] DataAccess.IEmployeeDal dal)
    {
        var data = dal.Get(EmployeeID);
        using (BypassPropertyChecks)
            Csla.Data.DataMapper.Map(data, this);
        BusinessRules.CheckRules();
    }

    [Fetch]
    private void Fetch(string firstname, string lastname, [Inject] DataAccess.IEmployeeDal dal)
    {
        var data = dal.Get(firstname, lastname);
        using (BypassPropertyChecks)
            Csla.Data.DataMapper.Map(data, this);
        BusinessRules.CheckRules();
    }

````

Your static method should not have the [Fetch] attribute. You can remove the static method and just call the DataPortal method with the required parameters.

**Comment by jhuerta3** [comment, marked as answer]
Thank you @JacoJordaan

**Reply by michaelcsikos** [reply]
@jhuerta3 With regard to what should be in your DAL interfaces, there are typical fetch / insert / update / delete methods, and then usually some others to populate drop-downs, grids, etc, for example:

```cs
public interface IPersonDal
{
    // Typical CRUD methods
    PersonDto Fetch   (Guid      id);
    void      Insert  (PersonDto dto);
    void      Update  (PersonDto dto);
    void      Delete  (Guid      id);

    bool      ExistsId(Guid      id);

    // The following can be useful for HTML Select, Table, etc,
    // used with CSLA read-only classes
    List<PersonNameDto> FetchPersonNameList(bool   includeArchived);

    List<PersonInfoDto> FetchList          (PersonTypeEnum type,
                                            bool           includeArchived,
                                            bool           includeNone,
                                            Guid?          companyId);

    List<PersonInfoDto> FetchList          (Guid?          excludeId,
                                            string         surname,
                                            string         firstName,
                                            string         emailAddress);

    // Sometimes you just need to fetch a name for one person
    PersonNameDto       FetchPersonName    (Guid   id);
}
```
